Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM node:18-alpine AS base
```

Comment:
this is probably something we want to keep track of with actions but it would be cool if we only rebuild certain layers and publish them to dockerhub only if there was a change in it's code tree

Reply:
that would be great,

for starters, thinking that inside actions we can
- use https://nx.dev/nx-api/nx/documents/affected to decide which targets to build and push

to only rebuild certain layers
- right now it builds everything in the builder step - need to figure out if there's a way to only build packages relevant to the target